pyscriptjs/src/plugins/pyterminal.ts

```diff
@@ -16,33 +29,31 @@ export class PyTerminalPlugin extends Plugin {
         this.app = app;
     }
 
-    configure(config: AppConfig & { terminal?: boolean | 'auto' }) {
+    configure(config: AppConfigStyle) {
         // validate the terminal config and handle default values
-        const t = config.terminal;
-        if (t !== undefined && t !== true && t !== false && t !== 'auto') {
-            const got = JSON.stringify(t);
-            throw new UserError(
-                ErrorCode.BAD_CONFIG,
-                'Invalid value for config.terminal: the only accepted' +
-                    `values are true, false and "auto", got "${got}".`,
-            );
-        }
+        const { terminal: t, docked: d } = config;
+        validate('terminal', 'auto', t);
+        validate('docked', 'docked', d);
         if (t === undefined) {
             config.terminal = 'auto'; // default value
         }
+        if (d === undefined) {
+            config.docked = 'docked'; // default value
+        }
```

Comment:
nitpick, but you could also add this logic to `validate`, couldn't you? So that `validate` returns the `default_` value in case you pass `undefined`. Something like this:
```javascript
config.termal = validate('terminal', 'auto', t);
config.docked = validate('docket', 'docket', d);
```

Reply:
I think I've simplified even more so no unnecessary re-assignment even if not needed happens and no need to destructure AOT neither.

Thanks for the hint though!